test/unit/lib/plugins/aws/deploy/lib/extendedValidate.test.js

```diff
@@ -193,5 +194,44 @@ describe('extendedValidate', () => {
         expect(awsDeploy.serverless.cli.log.firstCall.calledWithExactly(msg)).to.be.equal(true);
       });
     });
+
+    it("should not warn if function's timeout is greater than 30 and it's attached to APIGW, but it has [async] mode", () => {
+      stateFileMock.service.functions = {
+        first: {
+          timeout: 31,
+          package: {
+            artifact: 'artifact.zip',
+          },
+          events: [
+            {
+              http: {
+                async: true,
+              },
+            },
+          ],
+        },
+      };
+      awsDeploy.serverless.service.package.individually = true;
+      fileExistsSyncStub.returns(true);
+      readFileSyncStub.returns(stateFileMock);
+
+      return awsDeploy.extendedValidate().then(() => {
+        expect(awsDeploy.serverless.cli.log.firstCall).to.be.equal(null);
+      });
+    });
+
+    it("should not warn during the deployment if function's timeout is greater than 30 and it's attached to APIGW, but it has [async] mode", () => {
+      const msg = [
+        "WARNING: Function first has timeout of 31 seconds, however, it's",
+        "attached to API Gateway so it's automatically limited to 30 seconds.",
+      ].join(' ');
+
+      return runServerless({
```

Comment:
Let's use `async/await` instead of promise-based syntax as we're trying to migrate to `async/await` across the codebase